third_party/FMUParser/parser.c

```diff
@@ -38,292 +38,283 @@ FMU fmu;        // FMU to parse
 ///\param tmpPat Temporary path of the fmu file
 ///\return 0 if no error occurred
 ///////////////////////////////////////////////////////////////////////////
-int callparser(const char* fmuFilNam, const char* tmpPat){
-  size_t length;
-  char* xmlPat;
-  char* filNam, *tmp, *ext;
-
-  // Get the model description
-  xmlPat = calloc(sizeof(char), strlen(fmuFilNam) + strlen(XML_FILE)-1);
-  printfDebug("fmuFilNam is \"%s\"\n", fmuFilNam);
-  printfDebug("Start to parse model description file\n", xmlPat);
-  sprintf(xmlPat, "%s%s%s", tmpPat, PATH_SEP, XML_FILE);
-
-  fmu.modelDescription = parse(xmlPat);
-  printfDebug("Parsed model description file: \"%s\"\n", xmlPat);
+int callparser(const char *fmuFilNam, const char *tmpPat)
+{
+    size_t length;
+    char *xmlPat;
+    char *filNam, *tmp, *ext;
+
+    // Get the model description
+    xmlPat = (char *)calloc(strlen(fmuFilNam) + strlen(XML_FILE) + strlen(PATH_SEP) + 1, sizeof(char));
+    printfDebug("fmuFilNam is \"%s\"\n", fmuFilNam);
+    sprintf(xmlPat, "%s%s%s", tmpPat, PATH_SEP, XML_FILE);
+    printfDebug("Start to parse model description file: \"%s\".\n", xmlPat);
```

Comment:
Before was passing xmlPat to a statement with no placeholder... So move it after the sprintf and actually use it.